SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -4037,6 +4094,444 @@ void CEulerSolver::ReadActDisk_InputFile(CGeometry *geometry, CSolver **solver_c
   }
 }
 
+void CEulerSolver::SetActDisk_BEM_VLAD(CGeometry *geometry, CSolver **solver_container,
+                                       CConfig *config, unsigned short iMesh, bool Output) {
+
+  /*!
+   * \function SetActDisk_BEM_VLAD
+   * \brief Actuator disk model with Blade Element Method (BEM)
+   * \author: Chandukrishna Y., T. N. Venkatesh and Josy Pullockara
+   * Institution: Computational and Theoretical Fluid Dynamics (CTFD),
+   *            CSIR - National Aerospace Laboratories, Bangalore
+   *            Academy of Scientific and Innovative Research, Ghaziabad
+   * \version 8.0.0 "Harrier"
+   * First release date : September 26 2023
+   * modified on:
+   *
+   * Section properties of the propeller given in an input file.
+   * Cl, Cd of propeller sections need to be generated earlier and saved in this file
+   * Actuator disk data initialized in function SetActDisk_BCThrust.
+   * Propeller load calculated with Blade Element Method
+   * Interpolated load at each point on the actuator disk is set in SetActDisk_BEM_VLAD function
+   * Rest calculations follows the Variable Load (BC_ActDisk_VariableLoad) approach
+   *
+   */
+
+  /*--- InputFile reading ---*/
+  unsigned short iDim, iMarker;
+  unsigned long iVertex, iPoint;
+  string Marker_Tag;
+  int iSection, iAlpha;
+  static int ADBem_NBlade = 0, ADBem_NSection = 0, ADBem_NAlpha = 0;
+  static su2double ADBem_Diameter = 0.0, ADBem_HubRadius = 0.0, ADBem_Angle75R = 0.0;
+  static std::vector<su2double> i_v, radius_v, chord_v, angle75r_v;
+  static std::vector<std::vector<su2double> > alpha_m, cl_m, cd_m;
+
+  static su2double ADBem_Omega = 0.0;
+  static su2double ADBem_CG[MAXNDIM] = {0.0, 0.0, 0.0};
+  static su2double ADBem_Axis[MAXNDIM] = {0.0}, ADBem_J = 0.0;
+
+  /*--- BEM VLAD ---*/
+  const int BEM_MAX_ITER = 20;
+  su2double Omega_sw = 0.0, Omega_RPM = 0.0, Origin[3] = {0.0}, radius_[3] = {0.0};
+  static su2double omega_ref = 0.0, Lref = 0.0;
+  static su2double RPM = 0.0, blade_angle = 0.0, dia = 0.0, r_tip = 0.0, rps = 0.0, radius = 0.0;
+  static su2double V = 0.0, rho = 0.0, T = 0.0;
+  static su2double Torque = 0.0, dp_av = 0.0, dp_at_r = 0.0;
+  static std::vector<su2double> DtDr;
+  static su2double Target_Press_Jump = 0.0, Area = 0.0, UnitNormal[3] = {0.0}, Vn = 0.0;
+  static su2double Fa = 0.0, Ft = 0.0, Fr = 0.0, Fx = 0.0, Fy = 0.0, Fz = 0.0, dCp_v = 0.0, dCr_v = 0.0, rad_v = 0.0;
+  su2double *V_domain = 0, *Coord = 0;
+  static su2double loc_Torque = 0.0;
+  static su2double loc_thrust = 0.0;
+  static su2double tot_area = 0.0, tot_tq = 0.0;
+
+  unsigned long InnerIter = config->GetInnerIter();
+  su2double Dens_FreeStream = config->GetDensity_FreeStream();
+  const su2double* Vel_FreeStream = config->GetVelocity_FreeStream();
+
+  /*--- Input file provides force coefficients distributions along disk radius.
+        Initialization necessary only at initial iteration. ---*/
+  if (InnerIter == 0) {
```

Comment:
It's better to base the decision of reading the file on whether the tables have already been filed or not, e.g. check whether i_v is empty.